### PR TITLE
Add LDS01 driver

### PR DESCRIPTION
**Pull request by ecourtois**
Add LDS01 Lidar device driver using USART (without DMA) and an example.

### CODE REVIEW

File: drivers/include/lds01.h [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    char rx_mem[LDS01_BUFFER_SIZE];         /**< raw frame buffer */
+    ringbuffer_t rx_buf;                    /**< frame ringbuffer */
+    uint8_t remaining_bytes_in_frame;       /**< number of bytes to read for a full frame */
+} lds01_t;
+
+/**
+ * @brief Initialize LDS01 device
+ *
+ * @param[out]   lds01   lds01 device
+ * @param[in]   params  lds01 parameters
```
Comment (gdoffe) [inline drivers/include/lds01.h:97]: align lds01 and params

File: examples/lds01/include/lds01_params.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
```
Comment (gdoffe) [inline examples/lds01/include/lds01_params.h:1]: how the choice is made between this file and the default driver one ?

File: examples/lds01/main.c [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+int main(void)
+{
+    puts("\n== LDS01 example ==");
+
+    /* Start the frame reader thread */
+    frame_updater_thread_pid = thread_create(
+        frame_updater_thread_stack, sizeof(frame_updater_thread_stack),
+        FRAME_READER_PRIO, 0, frame_updater_thread, NULL, "frame_reader");
+
```
Comment (gdoffe) [inline examples/lds01/main.c:179]: 1 line is sufficient

File: drivers/include/lds01.h [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    driver_lds01 LDS01 (aka HLS-LFCD2) LiDAR driver
+ * @ingroup     drivers_sensors
+ * @brief       LDS01 (aka HLS-LFCD2) LiDAR driver
+ *
+ * The LDS01 (aka HLS-LFCD2) device is a 360° 2D LiDAR distance sensor.
```
Comment (Kheprys) [inline drivers/include/lds01.h:14]: LiDAR -> LIDAR

File: drivers/lds01/lds01.c [hunk 5]
```diff
@@ -0,0 +1,156 @@
+#include "lds01.h"
```
Comment (Kheprys) [inline drivers/lds01/lds01.c:1]: Global remark on this file - Using #define constants for LDS01 specifics values (number of value in one frame, length of frame, etc) could be great ease the read.

**Review by Kheprys — APPROVED** [review]
LGTM, a few comments though

File: drivers/include/lds01.h (hunk 1 above)
Comment (ecourtois) [reply drivers/include/lds01.h:97]: Done

File: examples/lds01/main.c (hunk 3 above)
Comment (ecourtois) [reply examples/lds01/main.c:179]: Done

File: drivers/include/lds01.h (hunk 4 above)
Comment (ecourtois) [reply drivers/include/lds01.h:14]: No it's really LiDAR ;-)

File: examples/lds01/include/lds01_params.h (hunk 2 above)
Comment (ecourtois) [reply examples/lds01/include/lds01_params.h:1]: Include directories from applications and examples are included before drivers includes (cf RIOT driver guidelines).

File: drivers/lds01/lds01.c (hunk 5 above)
Comment (ecourtois) [reply drivers/lds01/lds01.c:1]: Done

**Review by Kheprys — APPROVED** [review]

**Review by gdoffe — APPROVED** [review]

**Comment by gdoffe** [comment]
LGTM

**Pull request merged**